### PR TITLE
Adjusts cure chance for transformation viruses

### LINKED ISSUE #8568: Robotic transformation is incurable
Once contracted copper and spaceacillin don't do anything, the disease keeps progressing as normal. Impossible to escape from the transformation without killing the victim before it finishes its course. 

I'm guessing alien transformation might suffer from the same issue since it's another old disease but I couldn't get it to happen in my test server with xenomicrobes.

### PR DESCRIPTION
**Pull request by Intigracy**
- rscadd: "Curing transformation viruses you get from eating roburgers or autism spergetti should actually work now."

Went from 5% chance to 50% chance.

Before it was the lowest cure chance of any of the old shitty diseases.

fixes #8568

### CODE REVIEW

File: html/changelogs/Inticurevirus.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+author: Intigracy
+delete-after: True
+changes: 
+- rscadd: "Curing transformation viruses you get from eating roburgers or autism spergetti should actually work now."
```
Comment (PJB3005) [inline html/changelogs/Inticurevirus.yml]: more like tweak or bugfix.

**Comment by ghost** [comment]
Should become a lower and lower chance as it progresses, increasing the urgency of getting dat fukken cure?

File: html/changelogs/Inticurevirus.yml (hunk 1 above)
Comment (Intigracy) [inline html/changelogs/Inticurevirus.yml]: derp

**Comment by clusterfack** [comment]
Did you test this, really

**Comment by Intigracy** [comment]
Not even once, but I will rn.

@Skullyton I'm not rewriting old viro code.

**Comment by Intigracy** [comment]
Clusterfack I tested it and it didn't work.

Then I fixed the proc checking for cures because it was broken as shit.

File: code/datums/disease.dm [hunk 2]
```diff
@@ -78,27 +78,31 @@ var/list/diseases = typesof(/datum/disease) - /datum/disease
...
+				else
 					result = 0
-		else if(!affected_mob.reagents.has_reagent(cure_id))
+		else if(!affected_mob.reagents.has_reagent(cure_id)) //if cure id is just text, just check it against the reagents
 			result = 0
 	else
-		for(var/C_list in cure_list)
-			if(istype(C_list, /list))
-				for(var/C_id in cure_id)
-					if(affected_mob.reagents != null)
```
Comment (Intigracy) [inline code/datums/disease.dm:94]: If ya reagents aint null, ya obv aint got the cure!

**Comment by clusterfack** [comment]
Next time test your shit you fucking nerd

**Comment by Intigracy** [comment]
Testing is for nerds.

Thanks for asking though or I wouldn't even have bothered, I made the bad assumption that oldcode works.

**Comment by 9600bauds** [comment]
> I made the bad assumption that oldcode works.

We all made this mistake once

but it will still get you lynched someday

**Comment by Probe1** [comment]
> someday

File: code/datums/disease.dm [hunk 3]
```diff
@@ -78,27 +78,31 @@ var/list/diseases = typesof(/datum/disease) - /datum/disease
...
 /datum/disease/proc/has_cure()//check if affected_mob has required reagents.
-	if(!cure_id) return 0
+	if((!cure_id && !cure_list) || !curable) return 0
 	var/result = 1
-	if(cure_list == list(cure_id))
-		if(istype(cure_id, /list))
+	if(!cure_list || cure_list == cure_id) //if no cure_list or cure list is the same as cure_id, just check _id
+		if(istype(cure_id, /list)) //if cure_id is a list, check inside of it
 			for(var/C_id in cure_id)
-				if(!affected_mob.reagents.has_reagent(C_id))
```
Comment (clusterfack) [inline code/datums/disease.dm]: The original version of this loop made more sense, why are you setting result = 1 when it already is 1

Comment (Intigracy) [inline code/datums/disease.dm:86]: The original version of this loop was making result = 1 if you DIDN'T have the reagent to cure with, and 0 if you did.
I can remove the result = 1 though and just leave the break, w/e

Comment (Intigracy) [inline code/datums/disease.dm:86]: Wait nevermind, I added it for a reason.

This is the list where it's checking all of the reagents in that list and checking if ONE of them is in you. If it fails the first reagent but passes on the second, then it's going to have result = 0 without the result = 1 that's right here.

I could do a 

```
var/cures 
(for loop)
   if(reagent)
     cures++ 

if(!cures)
   result = 0
```

but is that better?

**Pull request merged**